node/pkg/driver/node.go

```diff
@@ -264,6 +263,36 @@ func (d *NodeService) nodeStageVolumeRequestValidation(req *csi.NodeStageVolumeR
 	return nil
 }
 
+func (d *NodeService) resolveFsTypeForMount(requestedFsType string, existingFormat string) (string, error) {
+	fsTypeForMount := requestedFsType
+	if requestedFsType == "" {
+		if existingFormat == "" {
+			fsTypeForMount = defaultFSType
+		} else {
+			fsTypeForMount = existingFormat
+		}
+	} else if existingFormat != "" {
+		if requestedFsType != existingFormat {
+			return "", status.Errorf(codes.AlreadyExists, "Requested fs_type {%v} but found {%v}", requestedFsType, existingFormat)
```

Comment:
no way of placing this in the validation part?

Reply:
sure, we could return `existingFormat` from the validation function, or just call `GetDiskFormat` twice.
though it would make us call `GetAccessType()` twice (at least it doesn't require an external i/o).
note that currently all validation functions only perform "static" checks (i.e. without external i/o, such as `GetDiskFormat`)

Reply:
ok, you right, better that  external i/o will be reduce as much as can be..